modules/core/src/main/java/org/locationtech/jts/geomgraph/GeometryGraph.java

```diff
@@ -292,6 +292,7 @@ private void addLineString(LineString line)
     if (coord.length < 2) {
       hasTooFewPoints = true;
       invalidPoint = coord[0];
+      addPoint(coord[0]);
```

Comment:
There should be a BIG comment here indicating that this is an enhancement to standard OGC semantics, and is potentially a source of (as yet unknown) logic bugs.